simulator-spring-boot/src/main/java/org/citrusframework/simulator/service/impl/DefaultScenarioExecutorServiceImpl.java

```diff
@@ -89,25 +99,33 @@ public final Long run(String name, @Nullable List<ScenarioParameter> scenarioPar
     public final Long run(SimulatorScenario scenario, String name, @Nullable List<ScenarioParameter> scenarioParameters) {
         logger.info("Starting scenario : {}", name);
 
-        ScenarioExecution scenarioExecution = scenarioExecutionService.createAndSaveExecutionScenario(name, scenarioParameters);
+        // Note that entity manager may not be closed here! It must stay open during the whole (asynchronous) transaction.
+        EntityManager entityManager = entityManagerFactory.createEntityManager();
+        EntityTransaction transaction = entityManager.getTransaction();
+        transaction.begin();
+
+        ScenarioExecution scenarioExecution = scenarioExecutionService.createAndSaveExecutionScenario(name, scenarioParameters, entityManager);
```

Comment:
now that I study this, I am pretty sure that the transaction must be committed first, then locked within a second transaction. the row wouldn't exits in the database otherwise. well, let's do an integration test first *sight 👀